lute/io/src/io.cpp

```diff
@@ -0,0 +1,141 @@
+#include "lute/io.h"
+#include "Luau/Variant.h"
+#include "lute/runtime.h"
+
+#include <uv.h>
+#include <memory>
+
+#include "lua.h"
+#include "lualib.h"
+
+
+namespace io
+{
+
+struct IOHandle
+{
+    Luau::Variant<uv_pipe_t, uv_tty_t> streamVariant;
+    uv_stream_t* stream = nullptr;
```

Comment:
I think we can get rid of this member and replace it with the getter method for streamVariant.